src/model/cfg.py

```diff
@@ -20,25 +23,120 @@ class DatasetType(Enum):
         "/media/omarabdelgawad/New Volume/Datasets/image_coloring/natural_view/"
     )
 
+# custom parser
+def positive_float(value: str) -> float:
+    ivalue = float(value)
+    if ivalue <= 0:
+        raise argparse.ArgumentTypeError(f"{value} is an invalid positive float value")
+    return ivalue
 
+def positive_int(value) -> int:
+    ivalue = int(value)
+    if ivalue <= 0:
+        raise argparse.ArgumentTypeError(f"{value} is an invalid positive int value")
+    return ivalue
+
+def custom_arg_parser(argv: Optional[Sequence[str]] = None) -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description="An unsupervised image-to-image translation model",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-lr", "--learning_rate",
+        type=positive_float,
+        default=2e-4,
+        help="Learning rate",
+        metavar="LEARNING_RATE",
+        dest="lr",
+    )
+    parser.add_argument(
+        "-b", "--batch_size",
+        type=positive_int,
+        default=16,
+        help="Batch size",
+        metavar="BATCH_SIZE",
+        dest="b",
+    )
+    parser.add_argument(
+        "-w", "--num_workers",
+        type=int,
+        default=2,
+        choices=[1, 2, 3, 4, 5, 6, 7, 8],
+        help="Number of workers",
+        metavar="NUM_WORKERS",
+        dest="w",
+    )
+    parser.add_argument(
+        "-i", "--image_size",
+        type=int,
+        default=256,
+        choices=[256, 512, 1024],
+        help="Image size",
+        metavar="IMAGE_SIZE",
+        dest="i",
+    )
+    parser.add_argument(
+        "-e", "--num_epochs",
+        type=positive_int,
+        default=100,
+        help="Number of epochs",
+        metavar="NUM_EPOCHS",
+        dest="e",
+    )
+    parser.add_argument(
+        "-l", "--load_model",
+        action="store_false",
+        default=True,
+        help="Load model",
+        dest="l",
+    )
+    parser.add_argument(
+        "-s", "--save_model",
+        action="store_true",
+        default=False,
+        help="Save model",
```

Comment:
We agreed that this argument should have a default value of True (maybe you swapped load and save unintentionally). For this one maybe use the following snippet
```python
parser.add_argument(
"--no-save-model",
action = "store_false"
default=True, # This line is not needed by the way.
help = "Don't save the trained model".
dest = "SAVE_MODEL"
)
```
Note that the dest name is still save_model and not inverted. That is because the args.SAVE_MODEL will contain the True by default and we want to save by default except if we specify not to.